pkg/types/openstack/platform.go

```diff
@@ -113,11 +113,17 @@ type Platform struct {
 	// +optional
 	IngressVIPs []string `json:"ingressVIPs,omitempty"`
 
-	// MachinesSubnet is the UUIDv4 of an openstack subnet. This subnet will be used by all nodes created by the installer.
+	// DeprecatedMachinesSubnet is a string of the UUIDv4 of an openstack subnet. This subnet will be used by all nodes created by the installer.
 	// By setting this, the installer will no longer create a network and subnet.
 	// The subnet and network specified in MachinesSubnet will not be deleted or modified by the installer.
 	// +optional
-	MachinesSubnet string `json:"machinesSubnet,omitempty"`
+	DeprecatedMachinesSubnet string `json:"machinesSubnet,omitempty"`
```

Comment:
Good point!
But given we're are converting the MachinesSubnet to ControlPlanePort because they represent the same thing, I believe it does not make sense to keep MachinesSubnet without mentioning deprecation of it even when in controlPlanePort is TP, but I'm no specialist here.
Note that with this PR the user can continue to define the MachinesSubnet in install-config normally.

@mandre @EmilienM any thoughts?

Reply:
Depends if we consider the API to be the json key `machinesSubnet`, or the field name `MachinesSubnet`. Ideally users will only interact with the json, but I suspect we also expect to fiddle with the structure directly as the fields are public?